packages/testsuite/cypress/e2e/mail/test-configuration-subsystem-mail-server.cy.ts

```diff
@@ -0,0 +1,207 @@
+describe("TESTS: Configuration => Subsystems => Mail", () => {
+  const address = [
+    "subsystem",
+    "mail",
+    "mail-session",
+    "my-mail-session",
+    "server",
+    "smtp",
+  ];
+
+  const outBoundSocketBinding = {
+    name: "custom-outbound-socket-binding",
+    host: "localhost",
+    port: "15099",
+  };
+
+  const serverForm = {
+    id: "mail-server-form",
+    outboundSocket: "outbound-socket-binding-ref",
+    userName: "username",
+    password: "password",
+    ssl: "ssl",
+    tls: "tls",
+  };
+
+  const addServerForm = {
+    id: "mail-server-add-form",
+    typeSelector: "#mail-server-add-form-server-type-editing",
+    outboundSocket: "outbound-socket-binding-ref",
+  };
+
+  const credentialsReferenceForm = {
+    id: "mail-server-credential-reference-add",
+    clearTextCliName: "credential-reference.clear-text",
+    clearText: "clear-text",
+  };
+
+  const testValues = {
+    outboundSocket: "custom-outbound-socket-binding",
+    mailSessionName: "my-mail-session",
+    userName: "Custom username",
+    password: "SecretPassword",
+    jndiName: "java:jboss/mail/MyJNDI",
+    clearText: "ClearTextValue",
+    serverType: "SMTP",
+  };
+
+  const mailSessionTable = {
+    id: "mail-server-table",
+    outboundSocketType: "SMTP",
+  };
+
+  const itemSelectors = {
+    mailServerItem: "#mail-server-item a",
+    mailServerAdd:
+      "div.pull-right.btn-group.hal-table-buttons > button:contains('Add')",
+    buttonAdd:
+      "div.modal-footer > button.btn.btn-hal.btn-primary:contains('Add')",
+    buttonYes:
+      "div.modal-footer > button.btn.btn-hal.btn-primary:contains('Yes')",
+    credentialsReferenceTab:
+      "#mail-server-tab-container > ul > li:nth-child(2) > a:contains('Credential Reference')",
```

Comment:
Is the `li:nth-child(2)` selector necessary? Wouldn't just `#mail-server-tab-container a:contains('Credential Reference')` work? I just think it would be best stay away from fixed index selectors (in case the UI would e.g switch orders of the navigation items)